code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -370,18 +370,20 @@ func (fsh *StorageHandler) RemoveCollaborator(ctx context.Context, r *http.Reque
 
 func (fsh *StorageHandler) GetFileStats(ctx context.Context, r *http.Request) (interface{}, error) {
 	allocationTx := ctx.Value(constants.ContextKeyAllocation).(string)
+	clientSign, _ := ctx.Value(constants.ContextKeyClientSignatureHeaderKey).(string)
+	clientPublicKey, _ := ctx.Value(constants.ContextKeyClientKey).(string)
```

Comment:
If you are using `clientPublicKey` from http header then it means `GetFileStats` is public endpoint and a subject to ddos attack.

If it is meant to be to only the owner of an allocation then we must use `allocationObj.OwnerPublicKey` in signature verification.

Reply:
Actually GetFileStats is a public endpoint whether I clientPublicKey from header or from allocationObj. It is a subject to DDoS attack in both cases, as we have seen in our DDoS tests. The difference is that getting it from allocationObj takes 60000 ms to respond (median response time), while getting it from header to fail the txns takes only ~100 ms (median response time). Moreover, security is also not an issue as whether we get it from header or we get it from allocationObj, we call verifySignatureFromRequest so we know if it's incorrect.